docs/usage.md

```diff
@@ -207,105 +255,326 @@ which are optional:
+that support templating. A variable is specified by a dot-separated path, with
+each component of the path representing a key in the YAML file. For example, if
+the version of the COS product appears as follows in the `product_vars.yaml`
+file:
+
+```yaml
+cos:
+  version: 2.4.76
+```
+
+This COS version can be used by specifying `cos.version` in a value in the
+input file.
+
+##### HPC CSM Software Recipe variable substitution example
+
+The following example bootprep input file shows how the COS version could be
+used in a bootprep input file that creates a CFS configuration for computes.
+Only one layer is shown for brevity.
+
+```yaml
+---
+configurations:
+- name: compute-{{recipe.version}}
+  layers:
+  - name: cos-compute-integration-{{cos.version}}
+    playbook: cos-compute.yaml
+    product:
+      name: cos
+      version: "{{cos.version}}"
+      branch: integration-{{cos.version}}
+```
+
+Note that when the value of a key in the bootprep input file is a Jinja2
+expression, it must be quoted to pass YAML syntax checking.
+
+Jinja2 expressions can also use filters and Python's built-in string methods to
+manipulate the variable values. For example, suppose only the major and minor
+components of the COS version are to be used in the branch name for the COS
+layer of the CFS configuration. The `split` string method can be used to
+achieve this as follows:
+
+```yaml
+---
+configurations:
+- name: compute-{{recipe.version}}
+  layers:
+  - name: cos-compute-integration-{{cos.version}}
+    playbook: cos-compute.yaml
+    product:
+      name: cos
+      version: "{{cos.version}}"
+      branch: integration-{{cos.version.split('.')[0]}}-{{cos.version.split('.')[1]}}
+```
+
+#### Dynamic variable substitutions
+
+In addition to the product version variables provided by the HPC CSM Software
+Recipe as described in [HPC CSM Software Recipe variable
+substitutions](#hpc-csm-software-recipe-variable-substitutions), additional
+variables are available. These variables are dynamic in that their values are
+determined at run-time based on the context in which they appear. The additional
+variables are as follows:
+
+- The variable `base.name` can be used in the `name` of an image in under the
+  `images` key. The value of this variable is the name of the IMS image or
+  recipe used as the base of this image.
+- The variable `image.name` can be used in the `name` of a session template
+  under the `session_templates` key. The value of this variable is the name of
+  the IMS image used in this session template.
+
+These variables reduce the need to duplicate values throughout the `sat
+bootprep` input file and also make the following use cases possible:
+
+- Using the name of the recipe in the name of an image built from a recipe
+  provided by a product.
+- Using the name of the image in the name of a session template that references an
+  image built from a recipe provided by a product.
```

Comment:
```suggestion
- You use the name of the recipe within the name of an image, and that image is
built from a recipe provided by a product.
- You use the name of the image within the name of a session template, and that
session template references an image that is built from a recipe provided by a product.
```

Reply:
I think the suggestion breaks the intended meaning a little bit. I'll think about how to reword.